jit-service/src/main/java/org/dcsa/jit/service/TransportCallService.java

```diff
@@ -34,15 +35,35 @@ public class TransportCallService {
   @Transactional
   public TransportCall ensureTransportCallExists(TimestampTO timestampTO) {
 
-    if ((timestampTO.importVoyageNumber() != null && timestampTO.exportVoyageNumber() != null)
-      == (timestampTO.carrierVoyageNumber() != null)) {
-      throw ConcreteRequestErrorMessageException.invalidInput(
-        "Cannot create timestamp where both voyage numbers carrierVoyageNumber AND (exportVoyageNumber + importVoyageNumber) " +
-          "is missing or provided together");
+    if (timestampTO.importVoyageNumber() != null || timestampTO.exportVoyageNumber() != null) {
+      if (timestampTO.importVoyageNumber() == null || timestampTO.exportVoyageNumber() == null) {
+        throw ConcreteRequestErrorMessageException.invalidInput(
+            "Both voyage exportVoyageNumber + importVoyageNumber should either be provided together or not at all"
+                + "is missing or provided together");
+      }
+      if (!(timestampTO.carrierVoyageNumber().equals(timestampTO.importVoyageNumber())
+          || timestampTO.carrierVoyageNumber().equals(timestampTO.exportVoyageNumber()))) {
+        throw ConcreteRequestErrorMessageException.invalidInput(
+            "When importVoyageNumber & exportVoyageNumber is given, then one of them has to equal the carrierVoyageNumber."
+                + "Please verify the values");
+      }
+    }
+
+    // Backwards compatibility with JIT 1.1
+    // if both exportVoyageNumber & importVoyageNumber are not given
+    // then we set them based on carrierVoyageNumber
+    // This is done to persist Voyage as carrierVoyageNumber is mandatory (Line 120)
+    if (timestampTO.importVoyageNumber() == null) {
```

Comment:
This is the same as saying both import & export is null, but there is no need to check both because if only one is null our previous test case catches it.